library(dplyr)
library(readxl)

# Lead SNPs gathering, generated by Chat-GPT #
lead_snps_auto <- read_excel("/maps/projects/kilpelainen-AUDIT/data/team_projects/Bora/input/clean_supplementary_tables/clean_satnumber_kulyte_suppl_2.xlsx")


# There are spaces between the numbers
lead_snps_auto <- read_excel("/maps/projects/kilpelainen-AUDIT/data/team_projects/Bora/input/clean_supplementary_tables/clean_satnumber_kulyte_suppl_2.xlsx") %>%
  mutate(
    POS = as.numeric(gsub(" ", "", POS)), # POS numberization
    P = as.numeric(gsub(",", ".", P))
  ) %>%
  arrange(CHR, POS) %>%
  group_by(CHR) %>%
  mutate(
    locus_group = cumsum(c(1, diff(POS) > 1e6))  # Lead SNP identification -Also done with the column lead/proxy
  ) %>%
  group_by(CHR, locus_group) %>%
  slice_min(P, n = 1, with_ties = FALSE) %>%
  ungroup()



colnames(lead_snps_auto)

lead_snps_auto <- lead_snps_auto %>%
  dplyr::select("CHR","POS","ID","REF","ALT","A1_FREQ","BETA","SE","P")

colnames(lead_snps_auto) <- c("chromosome","base_pair_location","variant","effect_allele","other_allele","effect_allele_frequency","beta","standard_error","p_value")


# Function application
sat <- pos_aligner_df(lead_snps_auto)
sat <- alleloi(sat)
sat <- eaf_chooser(sat)
sat <- mhc_cleaner(sat)

sat$sample_size <- NA

sat <- sat %>%
  mutate(chr_pos = paste(chromosome, base_pair_location, sep = ":"))

sat <- sat %>%
  mutate(chr_pos = paste0("chr",chromosome, ":",base_pair_location))

fwrite(sat, "/maps/projects/kilpelainen-AUDIT/data/team_projects/Bora/output/curated_supplementary_tables/satnumber_kulyte_curated.txt")
check <- fread("/maps/projects/kilpelainen-AUDIT/data/team_projects/Bora/output/curated_supplementary_tables/satnumber_kulyte_curated.txt")
